Model/Resource/Notification.php

```diff
@@ -36,21 +36,26 @@ public function _construct()
 
     /**
      * Get Notification for duplicate check
-     *
+     * 
      * @param $pspReference
      * @param $eventCode
      * @param $success
      * @param $originalReference
+     * @param null $done
      * @return array
      */
-    public function getNotification($pspReference, $eventCode, $success, $originalReference)
+    public function getNotification($pspReference, $eventCode, $success, $originalReference, $done = null)
     {
         $select = $this->getConnection()->select()
             ->from(['notification' => $this->getTable('adyen_notification')])
             ->where('notification.pspreference=?', $pspReference)
             ->where('notification.event_code=?', $eventCode)
             ->where('notification.success=?', $success);
 
+        if ($done) {
+            $select->where('notification.done=?', $done);
```

Comment:
this will not be executed if done=false

shall we change it to if($done===null) instead?

Reply:
False is not used in the code at the moment but agree better to check if it is null so we can support false in the future.